cmd/downloader/main.go

```diff
@@ -0,0 +1,127 @@
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/package-url/packageurl-go"
+
+	"github.com/ossf/package-analysis/internal/pkgmanager"
+	"github.com/ossf/package-analysis/internal/worker"
+	"github.com/ossf/package-analysis/pkg/api/pkgecosystem"
+)
+
+// Command-line tool to download a list of package archives, specified by purl
+// See https://github.com/package-url/purl-spec
+var (
+	packagesFile = flag.String("f", "", "file containing packages to download")
+	downloadDir  = flag.String("d", "", "directory to download files to (must exist)")
+)
+
+// usageError is a simple string error type, used when command usage
+// should be printed alongside the actual error message
+type cmdError struct {
+	message string
+}
+
+func (c *cmdError) Error() string {
+	return c.message
+}
+
+func newCmdError(message string) error {
+	return &cmdError{message}
+}
+
+func downloadPackage(purl packageurl.PackageURL, downloadDir string) error {
+	var ecosystem pkgecosystem.Ecosystem
+	if err := ecosystem.UnmarshalText([]byte(purl.Type)); err != nil {
+		return fmt.Errorf("unsupported package ecosystem for purl %s: %s", purl.String(), purl.Type)
+	}
+
+	manager := pkgmanager.Manager(ecosystem)
+	if manager == nil {
+		return fmt.Errorf("unsupported package ecosystem for purl %s: %s", purl.String(), purl.Type)
+	}
+
+	// Prepend package namespace to package name, if present
+	var pkgName string
+	if purl.Namespace != "" {
+		pkgName = purl.Namespace + "/" + purl.Name
+	} else {
+		pkgName = purl.Name
+	}
```

Comment:
Perhaps this logic can be moved into a function like ResolvePkg that returns our internal representation of a package when given a purl.

Reply:
Done